sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -0,0 +1,439 @@
+    for example in batch:
+      for key, tensor in example.items():
+        key_to_tensor_list[key].append(tensor)
+    key_to_batched_tensors = {}
+    for key in key_to_tensor_list:
+      batched_tensors = torch.stack(key_to_tensor_list[key])
+      batched_tensors = _convert_to_device(batched_tensors, device)
+      key_to_batched_tensors[key] = batched_tensors
+    predictions = model(**key_to_batched_tensors, **inference_args)
+    return utils._convert_to_result(batch, predictions, model_id)
+
+
+def _run_inference_tensorflow_keyed_tensor(
+    batch: Sequence[Dict[str, tf.Tensor]],
+    model: TFAutoModel,
+    device,
+    inference_args: Dict[str, Any],
+    model_id: Optional[str] = None) -> Iterable[PredictionResult]:
+  key_to_tensor_list = defaultdict(list)
+  for example in batch:
+    for key, tensor in example.items():
+      key_to_tensor_list[key].append(tensor)
+  key_to_batched_tensors = {}
+  for key in key_to_tensor_list:
+    batched_tensors = tf.stack(key_to_tensor_list[key], axis=0)
+    key_to_batched_tensors[key] = batched_tensors
+  predictions = model(**key_to_batched_tensors, **inference_args)
+  return utils._convert_to_result(batch, predictions, model_id)
+
+
+class HuggingFaceModelHandler(ModelHandler[ExampleT, PredictionT, ModelT], ABC):
+  def __init__(
+      self,
+      model_uri: str,
+      model_class: Union[AutoModel, TFAutoModel],
+      device: str = 'CPU',
+      *,
+      inference_fn: Union[
+          KeyedTensorInferenceFn,
+          TensorInferenceFn] = _run_inference_torch_keyed_tensor,
+      load_model_args: Optional[Dict[str, Any]] = None,
+      inference_args: Optional[Dict[str, Any]] = None,
+      min_batch_size: Optional[int] = None,
+      max_batch_size: Optional[int] = None,
+      large_model: bool = False,
+      **kwargs):
+    """Implementation of the abstract base class of ModelHandler interface
+    for Hugging Face. This class shouldn't be instantiated directly.
+    Use HuggingFaceModelHandlerKeyedTensor or HuggingFaceModelHandlerTensor.
+
+    Example Usage model::
+    pcoll | RunInference(HuggingFaceModelHandlerKeyedTensor(
+      model_uri="bert-base-uncased", model_class=AutoModelForMaskedLM))
+
+    Args:
+      model_uri (str): path to the pretrained model on the hugging face
+        models hub.
+      model_class: model class to load the repository from model_uri.
+      device: For torch tensors, specify device on which you wish to
+        run the model. Defaults to CPU.
+      inference_fn: the inference function to use during RunInference.
+        Default is _run_inference_torch_keyed_tensor or
+        _run_inference_tensorflow_keyed_tensor depending on the input type.
+      load_model_args (Dict[str, Any]): keyword arguments to provide load
+        options while loading from Hugging Face Hub. Defaults to None.
+      inference_args [Dict[str, Any]]: Non-batchable arguments
+        required as inputs to the model's forward() function. Unlike Tensors in
```

Comment:
forward is specific to `torch` framework. Can we make it more general like Arguments required to pass to the model's inference call.